### PR TITLE
Navigation Grid Cleanup

### PR DESCRIPTION
**Pull request by Crauzer**
Cleans up the Navigation Grid class

### CODE REVIEW

**Comment by danil179** [comment]
You should rebase your PR on the latest commit as there are conflicts.

**Comment by Crauzer** [comment]
PR is from latest commit not sure why NavGrid is conflicting since it is being removed

**Comment by danil179** [comment]
It's not. On your branch you are 4 commits behind.

**Comment by danil179** [comment]
Ping me when you finish rebasing on the last commit from LS (GitHub will also mark the changes in a useful way).

**Comment by Crauzer** [comment]
@danil179

File: GameServerLib/Content/Navigation/NavigationGrid.cs [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+        {
+            if (a is IObjBuilding)
+            {
+                double rayDist = Math.Sqrt(CastRaySqr(b.GetPosition(), a.GetPosition()));
+                rayDist += a.CollisionRadius;
+                return (rayDist * rayDist) < (b.GetPosition() - a.GetPosition()).SqrLength();
+            }
+            else
+            {
+                return CastRaySqr(a.GetPosition(), b.GetPosition()) < (b.GetPosition() - a.GetPosition()).SqrLength();
```
Comment (danil179) [inline GameServerLib/Content/Navigation/NavigationGrid.cs:590]: This section got changed. I think that your merge messed up some of the last commits. Try to re-rebase.

File: GameServerCore/Content/INavigationGrid.cs [hunk 2]
```diff
@@ -4,21 +4,16 @@
...
+        Vector3 MaxGridPosition { get; }
+        Vector3 TranslationMaxGridPosition { get; }
         float CellSize { get; }
-        uint XCellCount { get; }
-        uint YCellCount { get; }
+        uint CellCountX { get; }
+        uint CellCountY { get; }
         ushort[] CellFlags { get; } // Version 7 change
-        int XSampledHeightCount { get; }
-        int YSampledHeightCount { get; }
```
Comment (danil179) [inline GameServerCore/Content/INavigationGrid.cs:19]: I see that these variables get read in the code but not assigned to these class variables, why did you remove them?

File: GameServerLib/Content/Navigation/NavigationGridCell.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            this.RefHintWeight = br.ReadSingle();
+            br.ReadUInt16();   // <- "Unk2"
+            this.ArrivalDirection = br.ReadInt16();
+            this.RefHintNode[0] = br.ReadInt16();
+            this.RefHintNode[1] = br.ReadInt16();
+        }
+
+        public bool HasFlag(NavigationGrid grid, NavigationGridCellFlags flag)
+        {
+            return grid.HasFlag(this.ID, flag);
```
Comment (danil179) [inline GameServerLib/Content/Navigation/NavigationGridCell.cs:68]: I think that the flags of the NavGridCell should be saved in the cell and not in the NavGrid. Future issue.

**Review by danil179 — COMMENTED** [review]
I see mostly cosmetic changes, but something went wrong with your merge (you should rebase on LS indev from the first commit so the commits will not be in the commit list for this PR).

File: GameServerLib/Content/Navigation/NavigationGridCell.cs (hunk 3 above)
Comment (Crauzer) [reply GameServerLib/Content/Navigation/NavigationGridCell.cs:68]: I feel the same way.
Currently they are handled from the Navigation Grid itself which seems wrong.

File: GameServerCore/Content/INavigationGrid.cs (hunk 2 above)
Comment (Crauzer) [reply GameServerCore/Content/INavigationGrid.cs:19]: They aren't being used for anything.

File: GameServerLib/Content/Navigation/NavigationGrid.cs (hunk 1 above)
Comment (Crauzer) [reply GameServerLib/Content/Navigation/NavigationGrid.cs:590]: I did actually rebase and that didn't do anything so I merged from indev to my fork indev

Comment (danil179) [reply GameServerLib/Content/Navigation/NavigationGrid.cs:590]: The rebase worked. You can see that you actually have the right comments now, but with added merges that probably messed up the whole code. Try to remove these specific (and very carefully, probably add backup branch) commits that has already added in the rebase.

Comment (Crauzer) [reply GameServerLib/Content/Navigation/NavigationGrid.cs:590]: Should be correct now I think

Comment (danil179) [reply GameServerLib/Content/Navigation/NavigationGrid.cs:590]: More specifically, backup you branch and then `git rebase HEAD~10 -i` and then mark all the commits that has been already added with the rebase.

Comment (danil179) [reply GameServerLib/Content/Navigation/NavigationGrid.cs:590]: It's not correct now, see the `DashSpeed` variable that got removed.
